spec.emu

```diff
@@ -112,24 +110,50 @@ contributors: Chengzhong Wu, Hemanth HM, Ecma International
-              1. Let _msgDesc_ be the PropertyDescriptor { [[Value]]: _msg_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.
-              1. Perform ! DefinePropertyOrThrow(_O_, *"message"*, _msgDesc_).
-            1. <ins>If _cause_ is not *undefined*, then</ins>
-              1. <ins>Let _causeDesc_ be the PropertyDescriptor { [[Value]]: _cause_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</ins>
-              1. <ins>Perform ! DefinePropertyOrThrow(_O_, `"cause"`, _causeDesc_).</ins>
+              1. <del>Let _msgDesc_ be the PropertyDescriptor { [[Value]]: _msg_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</del>
+              1. <del>Perform ! DefinePropertyOrThrow(_O_, *"message"*, _msgDesc_).</del>
+              1. <ins>Perform ! CreateErrorObjectProperty(_O_, *"message"*, _msg_).</ins>
+            1. <ins>Perform ? ExtractErrorCause(_O_, _options_) </ins>
             1. Let _errorsList_ be ? IterableToList(_errors_).
             1. Perform ! DefinePropertyOrThrow(_O_, *"errors"*, PropertyDescriptor { [[Configurable]]: *true*, [[Enumerable]]: *false*, [[Writable]]: *true*, [[Value]]: ! CreateArrayFromList(_errorsList_) }).
             1. Return _O_.
           </emu-alg>
         </emu-clause>
       </emu-clause>
     </emu-clause>
+
+    <emu-clause id="sec-errorobjects-abstracts">
+      <h1>Abstract Operations for Error Objects</h1>
+
+      <emu-clause id="sec-errorobjects-create-property">
+        <h1><ins>CreateErrorObjectProperty ( _O_, _P_, _V_ )</ins></h1>
+        <p><ins>The abstract operation CreateErrorObjectProperty takes arguments _O_ (an Object), _P_ (a property key), and _V_ (an ECMAScript language value). It performs the following steps when called:</ins></p>
+        <emu-alg>
+          1. <ins>Assert: Type(_O_) is Object.</ins>
+          1. <ins>Assert: IsPropertyKey(_P_) is true.</ins>
+          1. <ins>Let _newDesc_ be the PropertyDescriptor { [[Value]]: _V_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</ins>
+          1. <ins>Return DefinePropertyOrThrow(_O_, _P_, _newDesc_).</ins>
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-errorobjects-extract-error-cause">
+        <h1><ins>ExtractErrorCause ( _O_, _options_ )</ins></h1>
+        <p><ins>The abstract operation ExtractErrorCause takes argument _O_ and _options_. It is used to create `"cause"` property on _O_ when the `"cause"` property is present on _options_. It performs the following steps when called:</ins></p>
+        <emu-alg>
+          1. <ins>Assert: Type(_O_) is Object.</ins>
+          1. <ins>If Type(_options_) is Object and ? HasProperty(_options_, `"cause"`) is *true*, then</ins>
+            1. <ins>Let _cause_ be ? Get(_options_, `"cause"`)</ins>
+            1. <ins>Perform ! CreateErrorObjectProperty(_O_, `"cause"`, _cause_).</ins>
+          1. <ins>Return *undefined*.</ins>
```

Comment:
is this needed at all?

Reply:
It's not required. I was assuming all abstract operations returning with values. If there are no such assumptions in ecma262, I'll remove this line.